pkg/pubsub/amqp/pooledsubscriber.go

```diff
@@ -75,17 +81,46 @@ func newPoolSubscriber(ctx context.Context, n uint, topic string, s subscriber,
 		topic:       topic,
 		msgChan:     msgChan,
 		poolMsgChan: poolMsgChan,
+		stopChan:    make(chan struct{}, 1),
```

Comment:
You do not need to use the buffered channel.
Changes you need to make:
1. `make(chan struct{}, 1)` - > `make(chan struct{})`
2. `s.stopChan <- struct{}{} ` - > `close(s.stopChan)`

Reply:
I need a buffer since the listener may already have been stopped by msgChan closing (this happens if the AMQP connection closes). In this case there will be no listener on the stopChan and then s.stopChan <- struct{}{} would hang.